src/modules/markets/actions/load-markets.js

```diff
@@ -197,7 +197,7 @@ export const loadMarketsByFilter = (filterOptions, cb = () => {}) => (
     }, 2000);
 
     // load categories here
-    dispatch(loadCategories(params));
+    dispatch(loadCategories(params, !params.category));
```

Comment:
This looks if a category is included in the filters. if it is tell redux store not to clear, just merge.